Sources/ProcedureKit/Procedure.swift

```diff
@@ -531,73 +767,171 @@ open class Procedure: Operation, ProcedureProtocol {
      Finish method which must be called eventually after an operation has
      begun executing, unless it is cancelled.
 
+     This method may not be overridden. To handle finishing, override
+     `procedureWillFinish(withErrors:)` or `procedureDidFinish(withErrors:)`
+     or use a WillFinish / DidFinish observer.
+
      - parameter errors: an array of `Error`, which defaults to empty.
      */
-    public final func finish(withErrors errors: [Error] = []) {
-        _finish(withErrors: errors, from: .finish)
+    public func finish(withErrors errors: [Error] = []) {
+        log.verbose(message: "finish() called")
+        finish(withErrors: errors, from: .finish)
+    }
+
+    // Used to queue an automatic finish from Procedure.start()/main()
+    // (i.e. if a Procedure should automatically finish prior to executing if, for example,
+    // it is cancelled prior to executing)
+    private func queueAutomaticFinish(from source: ProcedureKit.FinishingFrom) {
```

Comment:
Yep, because it's actually part of an internal struct `ProcedureKit` earlier in the file. (Has been organized like that for quite a while, I believe.)